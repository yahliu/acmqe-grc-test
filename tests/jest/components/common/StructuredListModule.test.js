/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import StructuredListModule from '../../../../src-web/components/common/StructuredListModule'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import * as reducers from '../../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import GrcApolloClient from '../../../../lib/client/apollo-client'
import { ApolloProvider } from 'react-apollo'
import { Provider } from 'react-redux'
import { staticResourceDataPolicyOverview, itemPolicyOverview } from './CommonTestingData'

describe('StructuredListModule component test', () => {
  it('renders as expected', () => {
    const preloadedState = window.__PRELOADED_STATE__
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), preloadedState, composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const staticResourceData = staticResourceDataPolicyOverview
    const component = renderer.create(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <Provider store={store}>
          <BrowserRouter>
            <StructuredListModule
              title={staticResourceData.detailKeys.title}
              headerRows={staticResourceData.detailKeys.headerRows}
              rows={staticResourceData.detailKeys.rows}
              data={[itemPolicyOverview]}
            />
          </BrowserRouter>
        </Provider>
      </ApolloProvider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
