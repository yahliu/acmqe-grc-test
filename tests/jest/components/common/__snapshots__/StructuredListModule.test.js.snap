// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StructuredListModule component test renders as expected 1`] = `
<div
  className="bx--module bx--module--double structured-list-module"
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header"
    >
      <h1
        className="bx--module__title"
      >
        Policy details
      </h1>
    </div>
    <div
      className="bx--module__content"
    >
      <section
        aria-label="Policy details"
        className="bx--structured-list bx--structured-list--condensed"
        role="region"
      >
        <div
          className="bx--structured-list-thead"
        >
          <div
            className="bx--structured-list-row bx--structured-list-row--header-row"
          >
            <div
              className="bx--structured-list-th"
            >
              Type
            </div>
            <div
              className="bx--structured-list-th"
            >
              Detail
            </div>
          </div>
        </div>
        <div
          className="bx--structured-list-tbody"
          onKeyDown={[Function]}
        >
          <div
            className="bx--structured-list-row"
          >
            <tr
              className="bx--parent-row-v2 bx--expandable-row-v2"
              data-parent-row={true}
            >
              <td
                className="bx--table-expand-v2"
                data-previous-value="collapsed"
                headers="expand"
              >
                <button
                  className="bx--table-expand-v2__button"
                  onClick={[Function]}
                >
                  <svg
                    alt="Provide a description that will be used as the title"
                    aria-label="Provide a description that will be used as the title"
                    className="bx--table-expand-v2__svg"
                    fillRule="evenodd"
                    height="12"
                    role="img"
                    viewBox="0 0 7 12"
                    width="7"
                  >
                    <title>
                      Provide a description that will be used as the title
                    </title>
                    <path
                      d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
                    />
                  </svg>
                </button>
              </td>
              <div
                className="bx--structured-list-td"
              >
                <p>
                  Name
                </p>
              </div>
              <div
                className="bx--structured-list-td"
              >
                <p>
                  case6-test-policy
                </p>
              </div>
            </tr>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
